src/zenml/client.py

```diff
@@ -7696,20 +7749,112 @@ def get_tag(
 
     def list_tags(
         self,
-        tag_filter_model: TagFilter,
+        sort_by: str = "created",
+        page: int = PAGINATION_STARTING_PAGE,
+        size: int = PAGE_SIZE_DEFAULT,
+        logical_operator: LogicalOperators = LogicalOperators.AND,
+        id: Optional[Union[UUID, str]] = None,
+        user: Optional[Union[UUID, str]] = None,
+        created: Optional[Union[datetime, str]] = None,
+        updated: Optional[Union[datetime, str]] = None,
+        name: Optional[str] = None,
+        color: Optional[Union[str, ColorVariants]] = None,
+        exclusive: Optional[bool] = None,
         hydrate: bool = False,
     ) -> Page[TagResponse]:
         """Get tags by filter.
 
         Args:
-            tag_filter_model: All filter parameters including pagination
-                params.
+            sort_by: The column to sort by.
+            page: The page of items.
+            size: The maximum size of all pages.
+            logical_operator: Which logical operator to use [and, or].
+            id: Use the id of stacks to filter by.
+            user: Use the user to filter by.
+            created: Use to filter by time of creation.
+            updated: Use the last updated date for filtering.
+            name: The name of the tag.
+            color: The color of the tag.
+            exclusive: Flag indicating whether the tag is exclusive.
             hydrate: Flag deciding whether to hydrate the output model(s)
                 by including metadata fields in the response.
 
         Returns:
             A page of all tags.
         """
         return self.zen_store.list_tags(
-            tag_filter_model=tag_filter_model, hydrate=hydrate
+            tag_filter_model=TagFilter(
+                sort_by=sort_by,
+                page=page,
+                size=size,
+                logical_operator=logical_operator,
+                id=id,
+                user=user,
+                created=created,
+                updated=updated,
+                name=name,
+                color=color,
+                exclusive=exclusive,
+            ),
+            hydrate=hydrate,
+        )
+
+    def attach_tag(
+        self,
+        tag_name_or_id: Union[str, UUID],
+        resources: List[TagResource],
+    ) -> None:
+        """Attach a tag to resources.
+
+        Args:
+            tag_name_or_id: name or id of the tag to be attached.
+            resources: the resources to attach the tag to.
+
+        Raises:
+            KeyError: If the tag is not found.
+        """
+        try:
```

Comment:
I'm hoping this method is not used during a pipeline run, since it is prone to race conditions. If it is, this needs to be refactored.

Reply:
We have implemented a lot of changes regarding the race conditions, feel free to check the changes. I am resolving this one for now, and I will keep the main thread in the `sql_zen_store` open for further discussion.